bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/mapper/SerialisedInputsToActionInputs.java

```diff
@@ -0,0 +1,179 @@
+                    }
+                } catch (IllegalArgumentException e) {
+                    failed = true;
+                }
+            } else {
+                try {
+                    return switch (input.getType()) {
+                        case "boolean", "java.lang.Boolean" -> Boolean.valueOf(valueString.toLowerCase());
+                        case "byte", "java.lang.Byte" -> Byte.valueOf(valueString);
+                        case "short", "java.lang.Short" -> Short.valueOf(valueString);
+                        case "int", "java.lang.Integer" -> Integer.valueOf(valueString);
+                        case "long", "java.lang.Long" -> Long.valueOf(valueString);
+                        case "float", "java.lang.Float" -> Float.valueOf(valueString);
+                        case "double", "java.lang.Double", "java.lang.Number" -> Double.valueOf(valueString);
+                        case "org.openhab.core.library.types.DecimalType" -> new DecimalType(valueString);
+                        case "java.time.LocalDate" ->
+                            // Accepted format is: 2007-12-03
+                            LocalDate.parse(valueString);
+                        case "java.time.LocalTime" ->
+                            // Accepted format is: 10:15:30
+                            LocalTime.parse(valueString);
+                        case "java.time.LocalDateTime" ->
+                            // Accepted format is: 2007-12-03 10:15:30
+                            LocalDateTime.parse(valueString, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+                        case "java.util.Date" ->
+                            // Accepted format is: 2007-12-03 10:15:30
+                            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse(valueString);
+                        case "java.time.ZonedDateTime" ->
+                            // Accepted format is: 2007-12-03T10:15:30+01:00[Europe/Paris]
+                            ZonedDateTime.parse(valueString);
+                        case "java.time.Instant" ->
+                            // Accepted format is: 2007-12-03T10:15:30.00Z
+                            Instant.parse(valueString);
+                        case "java.time.Duration" ->
+                            // Accepted format is: P2DT17H25M30.5S
+                            Duration.parse(valueString);
+                        default -> argument;
+                    };
+                } catch (NumberFormatException | DateTimeParseException | ParseException e) {
+                    failed = true;
+                }
+            }
+        }
+        if (failed) {
+            LOGGER.warn(
+                    "Action {} input parameter '{}': converting value '{}' into type {} failed! Input parameter is ignored.",
+                    actionType.getUID(), input.getName(), argument, input.getType());
+        }
+        return failed ? null : argument;
```

Comment:
see above: why don't you throw an exception in case of an error?

Reply:
IllegalArgumentException added